homeassistant/helpers/schema_config_entry_flow.py

```diff
@@ -182,6 +183,20 @@ async def _async_form_step(
                 self._options, user_input, data_schema
             )
 
+        if step_id == SOURCE_REAUTH and user_input is not None:
+            # Re-auth was validated; update the config entry, schedule reload and abort
+            hass = self.parent_handler.hass
+            entry = hass.config_entries.async_get_entry(
+                self.parent_handler.context["entry_id"]
+            )
+            assert entry
+            hass.config_entries.async_update_entry(
+                entry=entry,
+                options=user_input,
+            )
+            hass.config_entries.async_schedule_reload(entry.entry_id)
+            return self.parent_handler.async_abort(reason="reauth_successful")
```

Comment:
We have a helper for this:

```suggestion
            return self.parent_handler.async_update_reload_and_abort(
                entry, options=user_input
            )
```

Reply:
Thanks, I knew of this helper, but I was unsure if it is available there, my VSCode python plugin didn't like that I was fiddling around in these complex structures :sweat_smile:

Reply:
What if there are other options than what's in the reauth schema?
The helper takes care of that?

Reply:
Good question, I will check it tomorrow.

Reply:
The helper simply calls `async_update_entry`, `async_schedule_reload` and then `async_abort`

Reply:
> What if there are other options than what's in the reauth schema? The helper takes care of that?

I fixed it and updated the test, so that other options are preserved, thanks!